picky-server/src/http/controller.rs

```diff
@@ -313,6 +319,82 @@ impl ServerController {
 
         Ok(response)
     }
+
+    #[post("/ssh/sign_key")]
+    async fn sign_ssh_key(&self, req: Request) -> Result<ResponseBuilder, StatusCode> {
+        use serde::{Deserialize, Serialize};
+
+        #[derive(Deserialize, Serialize)]
+        struct SshSignRequest {
+            key: String,
+            key_type: SshKeyType,
+            key_id: String,
+            principals: Vec<String>,
+            duration_secs: u64,
+            nbf: u64,
+            exp: u64,
+        }
+
+        let sign_request: SshSignRequest = match check_authorization(&*self.read_conf().await, &req) {
+            Ok(token) => serde_json::from_value(token.claims).bad_request()?,
+            Err(e) => {
+                log::error!("authorization failed: {}", e);
+                return Err(StatusCode::UNAUTHORIZED);
+            }
+        };
+
+        let ssh_public_key = SshPublicKey::from_str(sign_request.key.as_str()).bad_request()?;
```

Comment:
What is wrong with `.as_str()`? Taking by a reference and calling `.as_str()` gives the same results. I prefer to call a function than use an operator in most cases.

Reply:
Oh, it's fine. I just myself prefer using the `Deref` semantic because it makes code concise and it's not less clear, but `as_str` is not wrong at all (I only use `as_str` when I specifically need a function such as `String -> &str` or if type inference fails for some reason). You may just mark this as resolved and change nothing, really.